File: CounterExample/Models/CounterModel.cs
using MvpCore.Events;
using MvpCore.Interfaces;
using CounterExample.Events;

namespace CounterExample.Models
{
    public class CounterModel : IModel
    {
        private int _count;
        private readonly IEventPublisher _eventBus;

        public CounterModel(IEventPublisher eventBus) => _eventBus = eventBus;

        public void Increment()
        {
            _count++;
            _eventBus.Publish(new CounterUpdatedEvent(_count));
        }
    }
}

File: counter-example/src/models/CounterModel.ts
import { IModel } from "mvp-core/interfaces/IModel";
import { IEventPublisher } from "mvp-core/interfaces/IEventPublisher";
import { CounterUpdatedEvent } from "../events/CounterEvents";

export class CounterModel implements IModel {
    private count = 0;

    constructor(private eventBus: IEventPublisher) { }

    increment(): void {
        console.log("increment");
        this.count++;
        this.eventBus.publish(new CounterUpdatedEvent(this.count));
    }
}